### PR TITLE
ParameterVector support for pulse parameter assignment

### LINKED ISSUE #5233: ParameterVector support for Schedules
### What is the expected enhancement?

Currently, `QuantumCircuit`s support `assign_parameters` with `ParameterVector` input. The task is to add support similarly for `Schedule`s, which now support parameter assignment as well.

### PR DESCRIPTION
**Pull request by arthurostrauss**
### Summary
This PR proposes an extended use of the current method `assign_parameters` in the `ParameterManager`to enable the assignment of a list of values through `ParameterVector`class. This enables an extended usage for the same methods in the `Schedule`and `ScheduleBlock`objects.

### Details and comments

The PR simply takes the already existing unrolling method from `QuantumCircuit`and applies it for the pulse level case, to enable a simple extension of the existing methods for parameter assignment.
This can be practical when mixing pulse level calibrations with many parameters.
This PR closes #5233

### CODE REVIEW

**Comment by arthurostrauss** [comment]
This PR would solve issue #5233

File: qiskit/pulse/parameter_manager.py
Comment (wshanks) [inline qiskit/pulse/parameter_manager.py]: Why does it check for `tuple` if the type hint only lists `list`? Perhaps everything should just be `collections.abc.Sequence`?

File: releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml
Comment (wshanks) [inline releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml]: I usually leave it to the release manager to decide what makes it into the prelude.

```suggestion
```

File: qiskit/pulse/parameter_manager.py
Comment (wshanks) [inline qiskit/pulse/parameter_manager.py]: Too bad there is no clean way to express paired key-value types on dicts.

File: releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml
Comment (wshanks) [inline releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml]:
```suggestion
features_pulse:
```

File: releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml
Comment (wshanks) [inline releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml]: This is good to note in the PR but doesn't need to be in the release notes since it is an internal helper function and the release notes are user facing.

```suggestion
```

File: qiskit/pulse/schedule.py
Comment (wshanks) [inline qiskit/pulse/schedule.py]: It was already using "Parameters" but I think here I would use "parameters" and "parameter vectors" unless you want to use Sphinx references to make it like `` :class:`.Parameter` ``s

**Review by wshanks — COMMENTED** [review]
This looks great! I had some feedback on the documentation, but the code looks fine.

File: qiskit/pulse/parameter_manager.py
Comment (arthurostrauss) [reply qiskit/pulse/parameter_manager.py]: The type is now typing.Sequence.

File: releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml
Comment (TsafrirA) [inline releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml]:
```suggestion
    the use of `ParameterVector` in pulse `Schedule` and `ScheduleBlock` objects when using their method
```

File: test/python/pulse/test_parameter_manager.py
Comment (TsafrirA) [inline test/python/pulse/test_parameter_manager.py:517]: For completeness, I think there should be a similar test with `Schedule`.

**Review by TsafrirA — CHANGES_REQUESTED** [review]
Looks good to me. Just a minor typo and perhaps an extra test.

**Comment by TsafrirA** [comment]
Thanks @arthurostrauss for the final fixes! This looks great.
@nkanazawa1989 - Why commits from other PRs are marked as differences here?

File: releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml
Comment (wshanks) [inline releasenotes/notes/pulse_parameter_manager_compat_with_ParameterVector-7d31395fd4019827.yaml]: This needs to be formatted as ReStructuredText rather than markdown. Here is I would phrase since it is a user-facing note and users do not generally work with `ParameterManager` directly.

```suggestion
features_pulse:
  - |
    The ``assign_parameters`` methods of :class:`.Schedule` and :class:`.ScheduleBlock`
    now support assigning a :class:`.ParameterVector` to a list of parameter values
    simultaneously in addition to assigning individual :class:`.Parameter` instances to
    individual values.
```

**Comment by wshanks** [comment]
I think GitHub is confused about the multiple branches/merges. When I look at the diff locally (like `git checkout -b tmp Qiskit/main` and `git merge --squash arthurostrauss/ParameterVector_pulse`), it looks right.

**Comment by wshanks** [comment]
@arthurostrauss This looks good to me now! Do you want to try to make it look clean in GitHub, so we know it will merge the way we want? I can squash things down and force push to your branch if you want.

**Comment by arthurostrauss** [comment]
> @arthurostrauss This looks good to me now! Do you want to try to make it look clean in GitHub, so we know it will merge the way we want? I can squash things down and force push to your branch if you want.

@wshanks Yes if you can do such force push (if necessary) I'd be happy to let you do it, as I'm not sure how to proceed on my side for this.

**Review by wshanks — APPROVED** [review]

**Comment by wshanks** [comment]
@arthurostrauss Actually, the diff looks good in GitHub now after the last merge, so I think it is fine to merge as is. I am not sure what messed up the diff view before.

**Pull request merged**
